File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.kt
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var max = 0
        var min = 0

        for(i in sizes.indices) {
            val sizeMax = if(sizes[i].first() >= sizes[i].last()) sizes[i].first() else sizes[i].last()
            val sizeMin = if(sizes[i].first() >= sizes[i].last()) sizes[i].last() else sizes[i].first()

            if(sizeMax > max) max = sizeMax
            if(sizeMin > min) min = sizeMin
        }

        return  max * min
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.swift
func solution(_ sizes:[[Int]]) -> Int {
    var maxSize = 0
    var minSize = 0
    
    for size in sizes {
        let max = size[0] > size[1] ? size[0] : size[1]
        let min = size[0] > size[1] ? size[1] : size[0]
        
        if maxSize < max {
            maxSize = max
            minSize = min
        }
    }
    
    for size in sizes {
        let min = size[0] > size[1] ? size[1] : size[0]
        
        if min > minSize {
            minSize = min
        }
    }

    return maxSize * minSize
}